services/src/main/scala/cromwell/services/instrumentation/impl/selectivetsv/SelectiveTsvInstrumentationServiceActor.scala

```diff
@@ -0,0 +1,154 @@
+
+/**
+  * Actor that ignores every InstrumentationServiceMessage - This is the default implementation of this service
+  */
+class SelectiveTsvInstrumentationServiceActor(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef)
+  extends Actor
+    with StrictLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+  context.system.scheduler.schedule(10.seconds, 1.seconds) { self ! SnapshotState }
+
+  var stateHistory: StateHistory = StateHistory.empty
+
+  override def receive: Receive = {
+    case InstrumentationServiceMessage(CromwellIncrement(CromwellBucket(_, path))) =>
+      val pathString = path.init.mkString(".")
+
+      if (path.last == "starting") {
+        stateHistory = stateHistory.increment(pathString)
+      } else if (path.last == "done") {
+        stateHistory = stateHistory.decrement(pathString)
+      }
+
+    case InstrumentationServiceMessage(CromwellGauge(CromwellBucket(_, path), value)) =>
+      val pathString = path.init.mkString(".")
+      if (path.last == "set") {
+        stateHistory = stateHistory.set(pathString, value)
+      }
+
+    case SnapshotState =>
+      stateHistory = stateHistory.snapshotState()
+      if (checkTsvPrintout()) {
+        stateHistory = StateHistory.empty
+        logger.info("Resetting state history")
+      }
+    //      logger.info(stateHistory.currentStateTsv())
+
+    case ShutdownCommand => context stop self
+
+    //    case other: InstrumentationServiceMessage => println(s"Ignoring $other")
+  }
+
+  private def checkTsvPrintout(): Boolean = {
+    if (stateHistory.stateHistory.size >= 2) {
+      if (stateHistory.stateHistory.last._2.get("jobs.ejea.executing").contains(0)) {
+        if (stateHistory.stateHistory.init.last._2.get("jobs.ejea.executing").exists(_ != 0)) {
+          outputCountHistory()
+          true
+        } else false
+      } else false
+    } else false
+  }
+
+  private def outputCountHistory(): Unit = {
+    import java.io.BufferedWriter
+    import java.io.FileOutputStream
+    import java.io.OutputStreamWriter
+    val fout = new File(s"selective-tsv-timestamps-${UUID.randomUUID().toString}.tsv")
+    val fos = new FileOutputStream(fout)
+
+    val bw = new BufferedWriter(new OutputStreamWriter(fos))
+
+    for (line <- stateHistory.stateHistoryTsv()) {
+      bw.write(line)
+      bw.newLine()
+    }
+    bw.flush()
+    bw.close()
+    logger.info(s"Printed out timestamps to ${fout.getAbsolutePath}")
+  }
+}
+
+object SelectiveTsvInstrumentationServiceActor {
+  def props(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef): Props =
+    Props(new SelectiveTsvInstrumentationServiceActor(serviceConfig, globalConfig, serviceRegistryActor))
+
+  case object SnapshotState
+
+  final case class StateHistory(fields: Vector[String], currentState: Map[String, Int], stateHistory: Vector[(Long, Map[String, Int])], firstEntryTimestampMillis: Option[Long]) {
```

Comment:
OOC why is `fields` maintained as a separate data structure from the keys of `currentState`?

Reply:
1. to keep the order consistent across the state histories
2. to make sure we zero-print fields which hadn't appeared yet in early histories so that their lines don't come out of nowhere when the first state history with a data point for them appearsl